p42/src/song.rs

```diff
@@ -0,0 +1,176 @@
+/// ```
+pub fn song_to_string(mut iter: impl Iterator<Item = String>) -> String {
+    let mut lyrics = iter.next().unwrap();
+    for line in iter {
+        lyrics.push('\n');
+        lyrics.push_str(line.as_str());
+    }
+    lyrics
+}
+
+/// extract lyrics from song iterator into file
+pub fn song_to_file(iter: impl Iterator<Item = String>, path: &String) -> std::io::Result<()> {
+    std::fs::write(path, song_to_string(iter))?;
```

Comment:
As I've mentioned during problem explanation, this function should not use `song_to_string` and instead should create `File` and write lines into it line-by-line.